pkg/chunk/encoding/chunk_test.go

```diff
@@ -28,26 +28,33 @@ import (
 )
 
 func TestLen(t *testing.T) {
-	chunks := []Chunk{}
 	for _, encoding := range []Encoding{Delta, DoubleDelta, Varbit} {
+		DefaultEncoding = encoding
+		chunks := []Chunk{}
 		c, err := NewForEncoding(encoding)
 		if err != nil {
 			t.Fatal(err)
 		}
 		chunks = append(chunks, c)
-	}
 
-	for _, c := range chunks {
 		for i := 0; i <= 10; i++ {
-			if c.Len() != i {
-				t.Errorf("chunk type %s should have %d samples, had %d", c.Encoding(), i, c.Len())
+			length := 0
+			for _, ch := range chunks {
+				length += ch.Len()
+			}
+			if length != i {
+				t.Errorf("chunk type %s should have %d samples, had %d", c.Encoding(), i, length)
 			}
 
-			cs, _ := c.Add(model.SamplePair{
+			nc, err := c.Add(model.SamplePair{
 				Timestamp: model.Time(i),
 				Value:     model.SampleValue(i),
 			})
-			c = cs[0]
+			require.NoError(t, err)
+			if nc != nil {
+				chunks = append(chunks, nc)
+				c = nc
+			}
```

Comment:
This test seems to have changed into a completely different test, and I don't understand why.  Not that it was a great test before, but it seems really twisted now.
If you still want to change it, can you write a sentence or two describing what it is doing?

Reply:
That was before I fixed my TODO for `transcodeAndAdd`. I brought back the test to the original state with some modifications for the new API.